src/Common/Config/ConfigProcessor.h

```diff
@@ -140,7 +140,7 @@ class ConfigProcessor
     static std::string decryptValue(const std::string & codec_name, const std::string & value);
 #endif
 
-    static inline const auto SUBSTITUTION_ATTRS = {"incl", "from_zk", "from_env"};
+    static inline const auto SUBSTITUTION_ATTRS = {"incl", "from_zk", "from_env", "from_vault"};
```

Comment:
We are adding a tag `from_vault` here. While the other terms `incl`, `from_zk`, `from_env` are ClickHouse-specific, `from_vault` isn't. We should at least make sure that the `vault` is a standard and accepted/used by dozens of companies. Is it?

If not, then something like `from_openbao_hashicorp_vault` would be nicer (or something similar).

Reply:
Your point is correct. I my point of view Hashicorp Vault is some kind of an industry standard used by many companies to keep secrets for web services. OpenBao is a fork of Hashicorp Vault aiming to continue keeping full API support.  

If you enter Vault in Google first you see Hashicorp Vault pages.

Top list of secrets managements services:

HashiCorp Vault, AWS Secrets Manager, Azure Key Vault, Google Secret Manager, Doppler, Infisical, CyberArk Conjur and Keeper Secrets Manager.

Thus 'vault' may be only confused with Azure Key Vault.

It's needed I may rename `from_vault` into `from_hashicorp_vault`, `from_openbao_vault` or `from_openbao_hashicorp_vault`. In this case it's not clear should `vault_key` also should be renamed?

Reply:
Let's then change to `from_hashicorp_vault` and `hashicorp_vault_key`.

Reply:
I think in this case section's name should be changed from `<vault>` to `<hashicorp_vault>` and also C++ class Vault should be changed to HashiCorpVault and other Python test related things with `with_vault` to `with_hashicorp_vault`. I am changing it now.